File: output/gemma4-e2b/avatar/Python/Java/atcoder_ABC168_B.java
import java.util.Scanner;

public class atcoder_ABC168_B {
    public static String main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // K = int(input())
        if (!scanner.hasNextInt()) {
            return ""; // Handle case where K is not provided
        }
        int K = scanner.nextInt();

        // S = input()
        if (!scanner.hasNext()) {
            return ""; // Handle case where S is not provided
        }
        String S = scanner.next();

        if (S.length() <= K) {
            return S;
        } else {
            // S[:K] + '...'
            return S.substring(0, K) + "...";
        }
    }
}

File: output/gemma4-e2b/avatar/Java/Python/atcoder_ABC168_B.py
import sys

def solve():
    # Read K
    try:
        # Read K from the first line
        K = int(sys.stdin.readline())
    except:
        # Handle case where input might be empty
        return

    # Read S from the second line
    S = sys.stdin.readline().strip()

    if len(S) <= K:
        print(S)
    else:
        # S.substring(0, K) corresponds to S[:K] in Python
        print(S[:K] + "...")

solve()